### ISSUE TITLE
[JENKINS-3265] Reload Configuration from Disk (or POSTing config.xml) loses info on running builds

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-21#2872]
<p>Executing the "Reload Configuration from Disk" while builds are running causes<br/>
information about those builds to be lost.</p>

<p>POSTing a new configuration to /job/JOBNAME/config.xml also causes the exact same problem. Yet editing job configuration interactively using the web form does not show the problem.</p>

<p>I had two builds running when I reloaded configuration to pick up manual changes<br/>
to one project (not running a build) config.xml file.  When the browser<br/>
refreshed, the builds were still shown as actively building in the Build<br/>
Executor Status frame but the status indicators were no longer blinking and the<br/>
project pages did not show any builds running for those projects.</p>

<p>It appears that part of Hudson thought that the builds had been canceled because<br/>
they were queued, blocked by themselves.  The original builds appeared to<br/>
finsish normally and then the queued builds ran.  There was no record of the<br/>
original build that had been running when the config reloaded.</p>

<p>Note: If Jenkins is completely restarted, by shutting down and starting up the service, the missing build re-appears. (Of course, doing that each time is unacceptable, just trying to help focus the debugging.)</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=pcampbell">pcampbell</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3265" target="_blank">Reload Configuration from Disk (or POSTing config.xml) loses info on running builds</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kohsuke.png" title="kohsuke's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kohsuke">kohsuke</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>component(s)</b>: core
<li><b>label(s)</b>: 1.509.1-fixed
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2015-02-11T12:40:15+00:00
<li><b>votes</b>: 28
<li><b>watchers</b>: 49
<li><b>imported</b>: 2025-11-22
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Executing the "Reload Configuration from Disk" while builds are running causes
information about those builds to be lost.</p>

<p>POSTing a new configuration to /job/JOBNAME/config.xml also causes the exact same problem. Yet editing job configuration interactively using the web form does not show the problem.</p>

<p>I had two builds running when I reloaded configuration to pick up manual changes
to one project (not running a build) config.xml file.  When the browser
refreshed, the builds were still shown as actively building in the Build
Executor Status frame but the status indicators were no longer blinking and the
project pages did not show any builds running for those projects.</p>

<p>It appears that part of Hudson thought that the builds had been canceled because
they were queued, blocked by themselves.  The original builds appeared to
finsish normally and then the queued builds ran.  There was no record of the
original build that had been running when the config reloaded.</p>

<p>Note: If Jenkins is completely restarted, by shutting down and starting up the service, the missing build re-appears. (Of course, doing that each time is unacceptable, just trying to help focus the debugging.)</p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>Platform: PC, OS: Windows XP</code></li></ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>samweiss:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3265?focusedId=143684&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-143684">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi</p>

<p>I too have this problem, and the way I manage my builds, this is getting very troublesome (I have automatic processes to create jobs for me for build branches that are created).</p>

<p>Any time Hudson reloads configuratons, not only are the info on running builds lost, you have to do another reload configuration once a running job is completed in order for hudson to recognize it.</p>

<p>Thanks!</p>

<p>Sam</p>
</pre>
</details>

</ul>
</details>

<p>Hi</p>

<p>I too have this problem, and the way I manage my builds, this is getting very troublesome (I have automatic processes to create jobs for me for build branches that are created).</p>

<p>Any time Hudson reloads configuratons, not only are the info on running builds lost, you have to do another reload configuration once a running job is completed in order for hudson to recognize it.</p>

<p>Thanks!</p>

<p>Sam</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/asull.jpg" title="asull's avatar" /> asull:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3265?focusedId=148010&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-148010">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>We are also struggling with this problem: it makes it problematic to use the reload from disk option because we have a lot of long-running (&gt;1hr) builds, and we don't want to lose visibility of currently executing jobs.</p>
</pre>
</details>

</ul>
</details>

<p>We are also struggling with this problem: it makes it problematic to use the reload from disk option because we have a lot of long-running (>1hr) builds, and we don't want to lose visibility of currently executing jobs.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>samweiss:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3265?focusedId=156123&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-156123">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Is there a status on this bug?  It was filed over a year ago, and still repros.  It's listed as "major"!  Hello????</p>
</pre>
</details>

</ul>
</details>

<p>Is there a status on this bug?  It was filed over a year ago, and still repros.  It's listed as "major"!  Hello????</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>pmv:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3265?focusedId=156748&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-156748">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>David - we also switched to using the cli to perform bulk changes.  I currently use the following groovy method to make a job isn't running before doing the update:</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java"><span class="code-keyword">public</span> <span class="code-object">boolean</span> isJobRunning(<span class="code-object">String</span> jobName) {
	URL jobDetails = <span class="code-keyword">new</span> URL(JENKINS_URL + <span class="code-quote">"/job/"</span> + jobName + <span class="code-quote">"/api/xml?depth=1"</span>)
	SAXParserFactory factory = SAXParserFactory.newInstance()
	factory.setFeature( <span class="code-quote">"http:<span class="code-comment">//apache.org/xml/features/nonvalidating/load-external-dtd"</span>, <span class="code-keyword">false</span>)
</span>	def details = <span class="code-keyword">new</span> XmlSlurper(factory.newSAXParser()).parse(jobDetails.openStream())
	<span class="code-object">String</span> building = details.build.building[0]
	<span class="code-keyword">if</span>(building.equals(<span class="code-quote">"<span class="code-keyword">true</span>"</span>)) {
		<span class="code-keyword">return</span> <span class="code-keyword">true</span>
	}
	<span class="code-keyword">return</span> <span class="code-keyword">false</span>
}
</pre>
</div></div> 

<p>Not really an answer to the original bug, but might help you...</p>
</pre>
</details>

</ul>
</details>

<p>David - we also switched to using the cli to perform bulk changes.  I currently use the following groovy method to make a job isn't running before doing the update:</p>


<pre>
<span class="code-keyword">public</span> <span class="code-object">boolean</span> isJobRunning(<span class="code-object">String</span> jobName) {
	URL jobDetails = <span class="code-keyword">new</span> URL(JENKINS_URL + <span class="code-quote">"/job/"</span> + jobName + <span class="code-quote">"/api/xml?depth=1"</span>)
	SAXParserFactory factory = SAXParserFactory.newInstance()
	factory.setFeature( <span class="code-quote">"http:<span class="code-comment">//apache.org/xml/features/nonvalidating/load-external-dtd"</span>, <span class="code-keyword">false</span>)
</span>	def details = <span class="code-keyword">new</span> XmlSlurper(factory.newSAXParser()).parse(jobDetails.openStream())
	<span class="code-object">String</span> building = details.build.building[0]
	<span class="code-keyword">if</span>(building.equals(<span class="code-quote">"<span class="code-keyword">true</span>"</span>)) {
		<span class="code-keyword">return</span> <span class="code-keyword">true</span>
	}
	<span class="code-keyword">return</span> <span class="code-keyword">false</span>
}
</pre> 

<p>Not really an answer to the original bug, but might help you...</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/nimeacuerdo.jpg" title="nimeacuerdo's avatar" /> nimeacuerdo:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3265?focusedId=156801&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-156801">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>First of all thanks for sharing, pmv.</p>

<p>The truth is that we switched to the cli tool when we found that we had to stop jenkins to perform bulk configuration changes on xmls due to the problem with reloading the config. But if we have to wait for jobs not to be running for the update to work from the cli, we are losing the advantage we wanted to get from using cli in the first place: the possibility of doing "hot" bulk updates of jobs configurations in a fast way.</p>

<p>Nevertheless I will have it in mind should we switch back again to the cli approach.</p>
</pre>
</details>

</ul>
</details>

<p>First of all thanks for sharing, pmv.</p>

<p>The truth is that we switched to the cli tool when we found that we had to stop jenkins to perform bulk configuration changes on xmls due to the problem with reloading the config. But if we have to wait for jobs not to be running for the update to work from the cli, we are losing the advantage we wanted to get from using cli in the first place: the possibility of doing "hot" bulk updates of jobs configurations in a fast way.</p>

<p>Nevertheless I will have it in mind should we switch back again to the cli approach.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jfrank.png" title="jfrank's avatar" /> jfrank:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3265?focusedId=157045&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-157045">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I am currently evaluating Jenkins and ran into this problem as well.
I must repeat samweiss's concern that this major bug has been open for a long time, approaching two or three years now (hmmm what is "10/Mar/09" is that 2009 or 2010? use four-digit years please). Is this typical for Jenkins bugs?
I added to the description a couple of things I found: POSTing config.xml triggers the same problem; editing the configuration via web form does not trigger the problem; and completely restarting the Jenkins service makes the disappeared jobs re-appear.
My workaround at the moment is to have my application (a .net console app) create a WebBrowser object to load the interactive form, adjust its content, and emulate a "click" on the submit button. It is ugly, especially since I have to mess around with thread apartment model to get WebBrowser to work from a console app, but it might let me get past this problem right now. Hope that can help a few people.</p>
</pre>
</details>

</ul>
</details>

<p>I am currently evaluating Jenkins and ran into this problem as well.<br/>
I must repeat samweiss's concern that this major bug has been open for a long time, approaching two or three years now (hmmm what is "10/Mar/09" is that 2009 or 2010? use four-digit years please). Is this typical for Jenkins bugs?<br/>
I added to the description a couple of things I found: POSTing config.xml triggers the same problem; editing the configuration via web form does not trigger the problem; and completely restarting the Jenkins service makes the disappeared jobs re-appear.<br/>
My workaround at the moment is to have my application (a .net console app) create a WebBrowser object to load the interactive form, adjust its content, and emulate a "click" on the submit button. It is ugly, especially since I have to mess around with thread apartment model to get WebBrowser to work from a console app, but it might let me get past this problem right now. Hope that can help a few people.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kutzi.jpg" title="kutzi's avatar" /> kutzi:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3265?focusedId=157047&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-157047">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Assigning to Kohsuke as I agree that this is a major issue which shouldn't stay open that long.</p>
</pre>
</details>

</ul>
</details>

<p>Assigning to Kohsuke as I agree that this is a major issue which shouldn't stay open that long.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>eekamak:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3265?focusedId=162673&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-162673">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>We were facing this problem when after Reload we wanted to trigger another jobs. I configured the Reload Job to be run as shell script like this and this seems to work:</p>

<p>curl <a href="http://localhost:80/reload" class="external-link" target="_blank" rel="nofollow noopener">http://localhost:80/reload</a>
sleep 25
curl <a href="http://localhost:80/job/$Job_name/build?delay=0sec" class="external-link" target="_blank" rel="nofollow noopener">http://localhost:80/job/$Job_name/build?delay=0sec</a></p>
</pre>
</details>

</ul>
</details>

<p>We were facing this problem when after Reload we wanted to trigger another jobs. I configured the Reload Job to be run as shell script like this and this seems to work:</p>

<p>curl <a href="http://localhost:80/reload" class="external-link" target="_blank" rel="nofollow noopener">http://localhost:80/reload</a><br/>
sleep 25<br/>
curl <a href="http://localhost:80/job/$Job_name/build?delay=0sec" class="external-link" target="_blank" rel="nofollow noopener">http://localhost:80/job/$Job_name/build?delay=0sec</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/antifun.png" title="antifun's avatar" /> antifun:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3265?focusedId=164010&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-164010">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I did some investigation into this, and here's what's going on:</p>

<ul>
	<li>The Reload Configuration operation from the UI attempts to load up all the jobs and runs it can find.</li>
	<li>It loads runs from disk based on the presence of a build.xml in the directory for a particular build number (cf. RunMap.load()).</li>
	<li>Runs that are in progress <b>do not</b> have a build.xml written to the directory in the course of normal operations (problem #1).</li>
</ul>


<p>Attempt 1: write a piece of code that, at the start of the reload operation (Jenkins.reload()), looks at all running Executors and forces them to marshal their state to disk (creating a build.xml). This would then allow Jenkins to notice those jobs running when it restarts.</p>

<p>Then I discovered that the unmarshal procedure for build.xml -&gt; Run assumes that any build.xml it sees is for a run that is in State.COMPLETED. The State is not itself persisted as there's no getState() for xstream to call.</p>

<p>Attempt 2: Make the State of a run persist appropriately so that it can be recovered when Jenkins reloads the configuration.</p>

<p>This seems to work OK at least in limited tests, and I intend to put up a pull request to let people see the changes I'm proposing in the code. I do wonder, though, what the justification for doing it this way in the first place was; it seems likely that you would <em>not</em> in all cases want Jenkins to totally "trust" the state on disk when starting up, for example if there were a large time or configuration delta between the stop and start. I do, however, think that the specific case of mashing the "Reload Configuration" link should be able to assume that what was running before is still running. Things that could possibly go wrong now would mostly be in the arena of jobs that Jenkins now thinks are running but actually aren't anymore. In a reload case, you can probably expect that to happen minimally (if at all).</p>

<p>Another idea I had was based on the fact that the executors clearly are reporting back the fact that they are running a particular job, even if Jenkins doesn't believe that run actually exists due to this bug. It might be possible/better to, instead of persisting everything to disk on the Jenkins master side, have it query the slaves for running jobs when it comes back up and use the information it gets from them to reconstruct its own idea of what is currently running. I don't know how people feel about potential for abuse there, given that it would require the master to "trust" the slaves to tell it what they were working on when it restarted. A combination of the two approaches might be best (trust, but verify).</p>
</pre>
</details>

</ul>
</details>

<p>I did some investigation into this, and here's what's going on:</p>

<ul>
	<li>The Reload Configuration operation from the UI attempts to load up all the jobs and runs it can find.</li>
	<li>It loads runs from disk based on the presence of a build.xml in the directory for a particular build number (cf. RunMap.load()).</li>
	<li>Runs that are in progress <b>do not</b> have a build.xml written to the directory in the course of normal operations (problem #1).</li>
</ul>


<p>Attempt 1: write a piece of code that, at the start of the reload operation (Jenkins.reload()), looks at all running Executors and forces them to marshal their state to disk (creating a build.xml). This would then allow Jenkins to notice those jobs running when it restarts.</p>

<p>Then I discovered that the unmarshal procedure for build.xml -> Run assumes that any build.xml it sees is for a run that is in State.COMPLETED. The State is not itself persisted as there's no getState() for xstream to call.</p>

<p>Attempt 2: Make the State of a run persist appropriately so that it can be recovered when Jenkins reloads the configuration.</p>

<p>This seems to work OK at least in limited tests, and I intend to put up a pull request to let people see the changes I'm proposing in the code. I do wonder, though, what the justification for doing it this way in the first place was; it seems likely that you would <em>not</em> in all cases want Jenkins to totally "trust" the state on disk when starting up, for example if there were a large time or configuration delta between the stop and start. I do, however, think that the specific case of mashing the "Reload Configuration" link should be able to assume that what was running before is still running. Things that could possibly go wrong now would mostly be in the arena of jobs that Jenkins now thinks are running but actually aren't anymore. In a reload case, you can probably expect that to happen minimally (if at all).</p>

<p>Another idea I had was based on the fact that the executors clearly are reporting back the fact that they are running a particular job, even if Jenkins doesn't believe that run actually exists due to this bug. It might be possible/better to, instead of persisting everything to disk on the Jenkins master side, have it query the slaves for running jobs when it comes back up and use the information it gets from them to reconstruct its own idea of what is currently running. I don't know how people feel about potential for abuse there, given that it would require the master to "trust" the slaves to tell it what they were working on when it restarted. A combination of the two approaches might be best (trust, but verify).</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/antifun.png" title="antifun's avatar" /> antifun:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3265?focusedId=164058&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-164058">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Change is here:</p>

<p><a href="https://github.com/antifun/jenkins/commit/2793ce3721de8c6a1e9d7912b92e437d6e827b2e" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/antifun/jenkins/commit/2793ce3721de8c6a1e9d7912b92e437d6e827b2e</a></p>
</pre>
</details>

</ul>
</details>

<p>Change is here:</p>

<p><a href="https://github.com/antifun/jenkins/commit/2793ce3721de8c6a1e9d7912b92e437d6e827b2e" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/antifun/jenkins/commit/2793ce3721de8c6a1e9d7912b92e437d6e827b2e</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/andrei_pozolotin.png" title="andrei_pozolotin's avatar" /> andrei_pozolotin:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3265?focusedId=164059&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-164059">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>possibly related: #<a href="https://issues.jenkins.io/browse/JENKINS-14123" title="rootPOM disappears on job edit in web interface" class="issue-link" data-issue-key="JENKINS-14123"><del>JENKINS-14123</del></a></p>
</pre>
</details>

</ul>
</details>

<p>possibly related: #<a href="https://issues.jenkins.io/browse/JENKINS-14123" title="rootPOM disappears on job edit in web interface" class="issue-link" data-issue-key="JENKINS-14123"><del>JENKINS-14123</del></a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jmiller.jpg" title="jmiller's avatar" /> jmiller:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3265?focusedId=181466&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-181466">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>We are still seeing this issue on 1.518 but only occasionally. Is anyone else seeing it?</p>

<p>Unfortunately we have not identified any particular characteristics of how or when this happens or if there is anything of note about the jobs it is happening with. If anyone has suggestions for things to look for to help diagnose the problem that would be most helpful.</p>
</pre>
</details>

</ul>
</details>

<p>We are still seeing this issue on 1.518 but only occasionally. Is anyone else seeing it?</p>

<p>Unfortunately we have not identified any particular characteristics of how or when this happens or if there is anything of note about the jobs it is happening with. If anyone has suggestions for things to look for to help diagnose the problem that would be most helpful.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/mpapo.png" title="mpapo's avatar" /> mpapo:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3265?focusedId=221576&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-221576">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi all,</p>

<p>Exactly the same problem : reload the config from disk cause current running builds to disappear after reload but the build directory is present on the disk.</p>

<p>Environment: Jenkins 1.593 on RHEL 6.0</p>
</pre>
</details>

</ul>
</details>

<p>Hi all,</p>

<p>Exactly the same problem : reload the config from disk cause current running builds to disappear after reload but the build directory is present on the disk.</p>

<p>Environment: Jenkins 1.593 on RHEL 6.0</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3265?focusedId=221590&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-221590">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Please keep issues as old as this resolved/closed.</p>

<p>So please file a new issue. Make sure it also happens on the current weekly release, as there've been possibly relevant changes since 1.593. Provide all the information asked for on <a href="https://wiki.jenkins-ci.org/display/JENKINS/How+to+report+an+issue" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/How+to+report+an+issue</a> and try to provide a minimal test case with clear instructions how to reproduce the problem.</p>
</pre>
</details>

</ul>
</details>

<p>Please keep issues as old as this resolved/closed.</p>

<p>So please file a new issue. Make sure it also happens on the current weekly release, as there've been possibly relevant changes since 1.593. Provide all the information asked for on <a href="https://wiki.jenkins-ci.org/display/JENKINS/How+to+report+an+issue" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/How+to+report+an+issue</a> and try to provide a minimal test case with clear instructions how to reproduce the problem.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/babul95.svg" title="babul95's avatar" /> babul95:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3265?focusedId=429460&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-429460">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>which path it refers to load configuration from local if we have 3 jenkins running on the same server</p>
</pre>
</details>

</ul>
</details>

<p>which path it refers to load configuration from local if we have 3 jenkins running on the same server</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `depends_on` from Jira: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-21/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-17337%22">JENKINS-17337</a>]</i>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `depends_on` from Jira: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-21/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-17341%22">JENKINS-17341</a>]</i>
